targets/csharp/make.js

```diff
@@ -262,36 +265,63 @@ function getAuthParams(apiCall, isApiInstance = false) {
     if (apiCall.auth === "SecretKey" && isApiInstance === true)
-        return tabbing + "var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? (authenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey);\n            if (developerSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"DeveloperSecretKey is not found in Request, Server Instance or PlayFabSettings\");\n";
+        return tabbing + "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API\n"
+			+ tabbing + "    var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? (authenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey);\n"
+			+ tabbing + "    if (developerSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"DeveloperSecretKey is not found in Request, Server Instance or PlayFabSettings\");\n"
+			+ tabbing + "#else\n"    
+			+ tabbing + "    var developerSecretKey = PlayFabSettings.DeveloperSecretKey;\n"
+			+ tabbing + "#endif\n";
     if (apiCall.url === "/Authentication/GetEntityToken")
-        return tabbing + "string authKey = null, authValue = null;\n"
-            + tabbing + "if (PlayFabSettings.ClientSessionTicket != null) { authKey = \"X-Authorization\"; authValue = PlayFabSettings.ClientSessionTicket; }\n"
-            + tabbing + "if (PlayFabSettings.DeveloperSecretKey != null) { authKey = \"X-SecretKey\"; authValue = PlayFabSettings.DeveloperSecretKey; }\n"
-            + tabbing + "if (PlayFabSettings.EntityToken != null) { authKey = \"X-EntityToken\"; authValue = PlayFabSettings.EntityToken; }\n";
+        return tabbing + "string authKey = null, authValue = null;\n" 
+            + tabbing + "#if !DISABLE_PLAYFABCLIENT_API\n"
+            + tabbing + "    var clientSessionTicket = request.AuthenticationContext?.ClientSessionTicket ?? PlayFabSettings.ClientSessionTicket;\n"
+            + tabbing + "    if (clientSessionTicket != null) { authKey = \"X-Authorization\"; authValue = clientSessionTicket; }\n"
+            + tabbing + "#endif\n"
+            + tabbing + "\n"
+            + tabbing + "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API\n"
+            + tabbing + "    var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey;\n"
+            + tabbing + "    if (developerSecretKey != null) { authKey = \"X-SecretKey\"; authValue = developerSecretKey; }\n"
+            + tabbing + "#endif\n"
+            + tabbing + "\n"
+            + tabbing + "#if !DISABLE_PLAYFABENTITY_API\n"
+            + tabbing + "    var entityToken = request.AuthenticationContext?.EntityToken ?? PlayFabSettings.EntityToken;\n"
+            + tabbing + "    if (entityToken != null) { authKey = \"X-EntityToken\"; authValue = entityToken; }\n"
+            + tabbing + "#endif\n";
```

Comment:
The #if 's and flags defined for GetEntityToken here are correct.
You DO need to add them here, because GetEntityToken is an Entity-Group method, but Client and Server may be enabled or disabled independently.
So, GetEntityToken logic is good and correct here (except for the invisible spacing on empty lines).